purchase_date_planned_manual/tests/test_date_planned_manual.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+from odoo import fields
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import UserError
+
+
+class TestDatePlannedManual(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        super(TestDatePlannedManual, self).setUp(*args, **kwargs)
+        self.po_model = self.env['purchase.order']
+        self.pol_model = self.env['purchase.order.line']
+        # Create some partner, products and supplier info:
+        route_buy_id = self.env.ref('purchase.route_warehouse0_buy').id
+        self.partner = self.env['res.partner'].create({
+            'name': 'Test supplier',
+            'supplier': True,
+        })
+        supplier_info_id = self.env['product.supplierinfo'].create({
+            'name': self.partner.id,
+            'delay': 2,
+        }).id
+        self.product_1 = self.env['product.product'].create({
+            'name': 'Test product 1',
+            'type': 'product',
+            'route_ids': [(4, route_buy_id)],
+            'seller_ids': [(4, supplier_info_id)],
+        })
+        self.product_2 = self.env['product.product'].create({
+            'name': 'Test product 2',
+            'type': 'product',
+        })
+        # Create purchase order and dates
+        self.purchase_order = self.env['purchase.order'].create({
```

Comment:
IMO `self.env['purchase.order']` replace to `self.po_model`

Reply:
Done